rotator/helpers.go

```diff
@@ -0,0 +1,165 @@
+package rotator
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	awsTools "github.com/mattermost/rotator/aws"
+	k8sTools "github.com/mattermost/rotator/k8s"
+	"github.com/mattermost/rotator/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	k8sErrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+// newNodes separates old nodes from new in a provided slice and returns new.
+func newNodes(allNodes, oldNodes []string) []string {
+	var newNodes []string
+	for _, node := range allNodes {
+		for _, oldNode := range oldNodes {
+			if node != oldNode {
+				newNodes = append(newNodes, node)
+			}
+		}
+	}
+	return newNodes
+}
```

Comment:
Amazing catch thanks a lot. This check is not that vital for the functionality because it is used to check that ASG nodes are up but it was doing duplicate work for no reason and I missed it. I changed to to a new function :)